bundles/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/handler/SpotifyDynamicStateDescriptionProvider.java

```diff
@@ -12,72 +12,61 @@
  */
 package org.openhab.binding.spotify.internal.handler;
 
-import static org.openhab.binding.spotify.internal.SpotifyBindingConstants.*;
-
-import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
-import java.util.Locale;
+import java.util.Map;
 import java.util.stream.Collectors;
 
-import org.eclipse.jdt.annotation.NonNull;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
-import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.binding.BaseDynamicStateDescriptionProvider;
 import org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionProvider;
-import org.eclipse.smarthome.core.types.StateDescription;
 import org.eclipse.smarthome.core.types.StateOption;
 import org.openhab.binding.spotify.internal.api.model.Device;
 import org.openhab.binding.spotify.internal.api.model.Playlist;
 import org.osgi.service.component.annotations.Component;
 
 /**
+ * Dynamically create the users list of devices and playlists.
  *
  * @author Hilbrand Bouwkamp - Initial contribution
  */
 @Component(service = { DynamicStateDescriptionProvider.class, SpotifyDynamicStateDescriptionProvider.class })
 @NonNullByDefault
-public class SpotifyDynamicStateDescriptionProvider implements DynamicStateDescriptionProvider {
+public class SpotifyDynamicStateDescriptionProvider extends BaseDynamicStateDescriptionProvider {
+
+    private Map<ChannelUID, @Nullable List<Device>> devicesByChannel = new HashMap<>();
```

Comment:
Both `Map`s can be final.

I recently discussed with @wborn that we do not want to add `@Nullable` to the value type of a `Map` because it is kind of a workaround to suppress the false positives of the compiler (see https://community.openhab.org/t/nonnullbydefault-and-hashmap/79845). Rather use `@SuppressWarnings("null")` on the `get()` method - which btw. is a workaround too but compliant with our guidelines.

Reply:
I've removed the `@Nullable`. However this will result in a warning on the null check in the set methods. The only way to suppress it is to set the `@SuppressWarnings("null")` on the set methods. But that suppresses it potential for the whole methods, and that doesn't seem right.

Reply:
I stumbled over that, too. It's not possible to suppress it for single instructions (at least in some cases).